packages/2019-disaster-game/src/state/tasks.js

```diff
@@ -74,6 +84,17 @@ export const tasksReducer = createReducer(initialState, {
   },
   [actionTypes.RESET_STATE]: () => {
     return initialState;
+  },
+  [actionTypes.COMPLETE_TASK]: (state, action) => {
+    const { id } = action.completedTask;
+
+    for (let i = 0; i < state.taskLocations.length; i += 1) {
+      const taskAtLocation = state.taskLocations[i];
+      if (taskAtLocation.type === id && !taskAtLocation.completed) {
+        state.taskLocations[i].completed = true;
+        break;
+      }
+    }
```

Comment:
Sorry, I just can't stop myself, I'm a monster 😞 

```js
const taskAtLocation = state.taskLocations.find(t => t.type === id); // find stops iterating once a match is found, so same as break
taskAtLocation.completed = true; // if it's already true, no harm in setting it to true anyway
```

Reply:
Don't have mental capacity to test this right now. Can only look forwards now